src/main/java/uk/gov/digital/ho/hocs/audit/service/SomuExportService.java

```diff
@@ -131,15 +133,26 @@ private String[] parseData(AuditEvent audit, List<SomuTypeField> headers,
         return data.toArray(new String[0]);
     }
 
-    private ExportDataConverter getDataConverter(boolean convert, String caseType) {
+    private ExportDataConverter getDataConverter(boolean convert, String caseType, List<SomuTypeField> somuTypeFields) {
         if (!convert) {
             return new ExportDataConverter();
         }
 
         Map<String, String> uuidToName = new HashMap<>(infoClient.getUsers().stream()
                 .collect(Collectors.toMap(UserDto::getId, UserDto::getUsername)));
 
-        return new ExportDataConverter(uuidToName, Collections.emptyMap(), caseType, auditRepository);
+        Map<String, String> entityListItemToName = new HashMap<>();
+        somuTypeFields.forEach(field -> {
```

Comment:
Are there any performance implications here?

Reply:
In short yes on the first run but we do cache these values.